### PR TITLE
fix(Scripts/Ulduar): reconstruct Freya's hard mode intro from sniff

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:
This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

Reconstructs Freya's hard mode intro (Elders alive at the pull) from a sniff. The sniffed sequence, reproduced here:

1. **Pull**: Freya casts Touch of Eonar and Attuned to Nature and yells. Each living Elder casts Full Heal (43978) on itself, becomes untargetable (`UNIT_FLAG_NOT_SELECTABLE`, flag update at pull +41ms in the sniff), enters combat silently, starts a real 5 second essence channel at Freya (62483/62484/62485, all three are 5s channeled spells in 3.3.5 DBC) and applies its banish state visual (Stonebark 62720 green, Ironbranch 62718 white, Brightleaf 61014 purple). Stonebark additionally casts Stonebark's Essence (62386, +50% physical damage) on Freya - the only essence buff visible in the sniff.
2. **Freya stands still** for those 5 seconds: her only movement packet at the pull is a facing-only spline, and her first real movement spline comes at +4.9s, exactly when the channels end. Implemented by disabling combat movement plus clearing the chase one tick after engage (the engaging `AttackStart` issues its `MoveChase` after `JustEngagedWith` returns) and parking the motion idle slot, since her default waypoint patrol (path 1365540) otherwise resumes underneath the cleared chase.
3. **Channel end (+5s)**: each Elder casts Drained of Power (62467) on itself. Its stun aura produces the `UNIT_FLAG_STUNNED` bit and root packets on its own (the sniff flag updates are timestamp-matched to the 62467 casts), so no flags are set manually for it. Freya resumes chasing at the same moment, matching the sniff.

Previously AC cast the essences and Drained of Power instantly at the pull (no channels, no untargetability, no hold) and Freya immediately ran at the raid. The Elders also played their solo-encounter aggro yells when dragged into combat this way; the sniff contains zero elder chat at the pull, so they are silenced (and their miniboss ability schedules skipped) when pulled via hard mode activation. Wipe handling: Freya's reset force-evades the Elders, which clears the auras, and their `Reset` clears the untargetable flag.

### AI-assisted Pull Requests

> [!IMPORTANT]
> Using AI tools to prepare pull requests is allowed, but it must be disclosed and it must follow our AC guidelines for AI Agentic Engineering (link below).
>
> You are expected to fully understand the changes you submit and to be able to explain and justify them when maintainers ask.

- [x] AI tools (e.g. Claude, ChatGPT, or similar) were used entirely or partially to prepare this pull request. If checked, specify which tools and models below.
   - Tools/models used: Claude Code (Claude Fable 5)
- [x] I have read and understood the [AC guidelines for AI Agentic Engineering](https://www.azerothcore.org/wiki/agentic-engineering)

## Issues Addressed:
- None

## SOURCE:
The changes have been validated through:
- [x] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [x] Sniffs (remember to share them with the open source community!)
- [x] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

Sniff of a hard mode Freya kill with all three Elders alive. Spell data cross-checked on Wowhead's 3.3.5 database (essence channels are Channeled/5s; Drained of Power is Apply Aura: Stun; banish states and Full Heal all exist as 3.3.5 spells).

## Tests Performed:
This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

Builds without errors (Windows, MSVC Release). Tested in-game: hold works from any point of her patrol, Elders stay silent and untargetable, Drained lands at channel end.

## How to Test the Changes:
- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [x] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1. Pull Freya with all three Elders alive. She must yell, stand in place facing her target for ~5 seconds while each Elder channels a beam at her, then start attacking.
2. During those 5 seconds the Elders must be untargetable, silent (no aggro yells) and cast nothing offensive; at the end they get Drained of Power and stay stunned/untargetable.
3. With Stonebark alive, Freya's melee should hit noticeably harder (Stonebark's Essence, +50% physical damage) - this was silently missing before, so tank damage goes up in hard mode.
4. Wipe and re-pull: intro repeats cleanly; after a wipe the Elders are targetable and normal again.
5. Kill the Elders first and pull Freya: normal aggro yell, no hold, she engages immediately.
6. Pull an Elder on its own (no Freya): it must still yell and fight normally.

## Known Issues and TODO List:

- [ ] Only Stonebark's essence buff (62386) is visible in the sniff; if Ironbranch/Brightleaf have equivalent stat buffs they could not be identified from this data.
- [ ] The Elders drop their in-combat flag ~9s after the pull on retail; on AC they stay in zone combat until wipe/kill (cosmetic difference).

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: src/server/scripts/Northrend/Ulduar/Ulduar/boss_freya.cpp [hunk 1]
```diff
@@ -473,39 +479,72 @@ struct boss_freya : public BossAI
...
+            for (uint8 i = 0; i < 3; ++i)
+            {
+                elder = ObjectAccessor::GetCreature(*me, _elderGUID[i]);
+                if (!elder)
+                    continue;
+
+                elder->CastSpell(elder, SPELL_FULL_HEAL, true);
+                elder->SetUnitFlag(UNIT_FLAG_NOT_SELECTABLE);
+                elder->SetInCombatWithZone();
+                elder->CastSpell(me, essenceChannel[i], false);
```
[the remark this answers is not in the document]
Comment (Nyeriah) [reply src/server/scripts/Northrend/Ulduar/Ulduar/boss_freya.cpp:519]: took the suggestion, they're REACT_PASSIVE for the intro now and their Reset restores aggressive for solo pulls. In practice the channel already blocked chase and swings via the casting state, but this covers the channel-end boundary tick too

**Review by Gultask — APPROVED** [review]

**Comment by Nyeriah** [comment]
Self-review turned up one edge case that this PR does not cover, recording it here rather than expanding the change.

If an Elder is already in combat when Freya is pulled (raid engages Ironbranch, someone then aggros Freya while it is still up), that Elder keeps running its miniboss abilities for the whole 5 second intro. The new `JustEngagedWith` guard only suppresses scheduling for Elders that the hard mode activation itself drags into combat. One that engaged earlier already has its events queued, and [`CreatureAI::UpdateVictim`](https://github.com/azerothcore/azerothcore-wotlk/blob/7eea30d471966d970c0989c11ccc85110d064a17/src/server/game/AI/CreatureAI.cpp#L372-L380) returns true for a passive creature that is in combat, so its `UpdateAI` keeps executing them. Thorn Swarm still picks a player off the threat list.

Before this PR the Elder was stunned by Drained of Power at the instant of the pull, so the window is new, and it lasts the 5 seconds until Drained lands.

Closing it is a one-liner: `elder->AI()->Reset();` as the first statement of the loop body, ahead of the `SetUnitFlag` call. Reset clears the stale EventMap, and the lines below it re-apply the flag and react state it undoes. Order matters, placed after `SetUnitFlag` it would wipe the untargetable flag.

Happy to add it if a maintainer would rather have it in this PR.

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by Nyeriah** [comment]
> If an Elder is already in combat when Freya is pulled (raid engages Ironbranch, someone then aggros Freya while it is still up), that Elder keeps running its miniboss abilities for the whole 5 second intro. The new JustEngagedWith guard only suppresses scheduling for Elders that the hard mode activation itself drags into combat. One that engaged earlier already has its events queued, and [CreatureAI::UpdateVictim](https://github.com/azerothcore/azerothcore-wotlk/blob/7eea30d471966d970c0989c11ccc85110d064a17/src/server/game/AI/CreatureAI.cpp#L372-L380) returns true for a passive creature that is in combat, so its UpdateAI keeps executing them. Thorn Swarm still picks a player off the threat list.

Tested it on offy, and they just get stunned wherever they happen to me at the moment the spell hits them, no change required here
